File: reto_clean_code/python/domain/producto.py
"""
Modelo de dominio para un producto dentro del catálogo.  Incluye atributos
básicos y un método para actualizar el stock.
"""

from dataclasses import dataclass


@dataclass
class Producto:
    id: str
    nombre: str
    precio: float
    stock: int

    def actualizar_stock(self, nueva_cantidad: int) -> None:
        self.stock = nueva_cantidad

File: reto_clean_code/angular/domain/producto.ts
// Definición de un producto en el dominio de un e‑commerce.  Incluye
// propiedades básicas y un método para actualizar el stock.

export class Producto {
  constructor(
    public id: string,
    public nombre: string,
    public precio: number,
    public stock: number
  ) {}

  actualizarStock(nuevaCantidad: number): void {
    this.stock = nuevaCantidad;
  }
}